sormas-backend/src/test/java/de/symeda/sormas/backend/visit/ExternalVisitTest.java

```diff
@@ -0,0 +1,483 @@
+
+		// Define relevant changes
+		HashMap<String, Object> relevantChanges = new HashMap<String, Object>() {
+
+			{
+				put(Person.FIRST_NAME, "Heinz");
+				put(Person.LAST_NAME, "Müller");
+				put(Person.SEX, Sex.FEMALE);
+				put(Person.BIRTHDATE_YYYY, 2001);
+				put(Person.BIRTHDATE_MM, 7);
+				put(Person.BIRTHDATE_DD, 2);
+			}
+		};
+
+		person.setPhone("+496211218491");
+		person.setEmailAddress("heinz@test.de");
+
+		// Apply each change and make sure it makes notification considered necessary
+		for (String propertyName : relevantChanges.keySet()) {
+			journalPerson = personFacade.getPersonForJournal(person.getUuid());
+			setPersonProperty(person, propertyName, relevantChanges.get(propertyName));
+			person = entityManager.merge(person);
+			assertTrue(getExternalJournalService().notifyExternalJournalPersonUpdate(journalPerson).getElement0());
+
+			// Modify the SymptomJournalStatus of the original person
+			journalPerson = personFacade.getPersonForJournal(person.getUuid());
+			person.setSymptomJournalStatus(SymptomJournalStatus.DELETED);
+			person = entityManager.merge(person);
+			assertFalse(getExternalJournalService().notifyExternalJournalPersonUpdate(journalPerson).getElement0());
+
+			journalPerson = personFacade.getPersonForJournal(person.getUuid());
+			person.setSymptomJournalStatus(SymptomJournalStatus.REJECTED);
+			person = entityManager.merge(person);
+			assertFalse(getExternalJournalService().notifyExternalJournalPersonUpdate(journalPerson).getElement0());
+
+			journalPerson = personFacade.getPersonForJournal(person.getUuid());
+			person.setSymptomJournalStatus(SymptomJournalStatus.UNREGISTERED);
+			person = entityManager.merge(person);
+			assertFalse(getExternalJournalService().notifyExternalJournalPersonUpdate(journalPerson).getElement0());
+
+			journalPerson = personFacade.getPersonForJournal(person.getUuid());
+			person.setSymptomJournalStatus(SymptomJournalStatus.ACCEPTED);
+			person = entityManager.merge(person);
+			assertFalse(getExternalJournalService().notifyExternalJournalPersonUpdate(journalPerson).getElement0());
+
+			// Apply any other relevant change and make sure notification is still considered necessary
+			for (String secondPropertyName : relevantChanges.keySet()) {
+				if (!secondPropertyName.equals(propertyName)) {
+					journalPerson = personFacade.getPersonForJournal(person.getUuid());
+					setPersonProperty(person, secondPropertyName, relevantChanges.get(secondPropertyName));
+					person = entityManager.merge(person);
+					assertTrue(getExternalJournalService().notifyExternalJournalPersonUpdate(journalPerson).getElement0());
+				}
+			}
+
+			setPersonRelevantFields(person);
+			person = entityManager.merge(person);
+		}
+	}
+
+	@Test
+	public void testCreateExternalVisit() {
+		PersonDto cazePerson = creator.createPerson("Case", "Person");
+		CaseDataDto caze = creator.createCase(
+			externalVisitsUser.toReference(),
```

Comment:
Maybe it's a more realistic approach to create the entities with a different user (e.g. nationalUser). The externalVisitsUser usually does not create them. This comment applies not only to this line, but to every entity creation in this test class.

Reply:
those entities are created with the test user, the external visit is used only for calling the method in test e.g. 
```
loginWith(externalVisitsUser);

visitFacade.saveExternalVisit(externalVisitDto);
visitFacade.saveExternalVisit(externalVisitDto2);
```